File: Frontend/src/types/analytics.ts
/* Analytics-related types. */

export interface AnalyticsOverview {
  total_clicks: number;
  clicks_today: number;
  top_links: TopLink[];
}

export interface TopLink {
  title: string;
  clicks: number;
}

export interface HeatmapData {
  hour: number;
  clicks: number;
}

export interface TrafficSourceData {
  source: string;
  clicks: number;
}

File: Backend/app/schemas/analytics.py
"""Analytics schemas."""

from pydantic import BaseModel
from uuid import UUID
from typing import Optional


class ClickRecordRequest(BaseModel):
    link_id: UUID
    referrer: Optional[str] = None
    country: Optional[str] = None


class TopLinkItem(BaseModel):
    title: str
    clicks: int


class AnalyticsOverview(BaseModel):
    total_clicks: int
    clicks_today: int
    top_links: list[TopLinkItem]


class HeatmapData(BaseModel):
    hour: int
    clicks: int


class TrafficSourceData(BaseModel):
    source: str
    clicks: int
